### PR TITLE
Analyzers code and docs reorganization

### PR DESCRIPTION
**Pull request by gao-artur**
This PR just makes some code reorganization, adds documentation to extension methods, and reformats markdown files to break line after ~80 column. No changes in logic.

I used [Prettier](https://prettier.io/) extension for VS Code to automatically break lines. But it modified [packages files](https://github.com/graphql-dotnet/graphql-dotnet/commit/087b777aa0de1eb02ed92a77a0029ad248967848). I have no idea what are these files, so let me know if you want to revert this change. Or if you want to move the `.prettierrc.json` file to the root docs folder.

### CODE REVIEW

File: src/GraphQL.sln.DotSettings [hunk 1]
```diff
@@ -0,0 +1,2 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
+	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/Abbreviations/=QL/@EntryIndexedValue">QL</s:String></wpf:ResourceDictionary>
```
Comment (Shane32) [inline src/GraphQL.sln.DotSettings:2]: What's this for?  User preferences or project settings?

Comment (gao-artur) [reply src/GraphQL.sln.DotSettings:2]: This is ReSharper config file to stop complaining about QL in capital letters. The config says it's abbreviation.

**Review by Shane32 — APPROVED** [review]

File: src/GraphQL.sln.DotSettings (hunk 1 above)
Comment (gao-artur) [reply src/GraphQL.sln.DotSettings:2]: I'll revert this file. And also packages. It looks like CI doesn't like these changes.

**Comment by Shane32** [comment]
Tests fail so have to check that but looks good in general

**Pull request merged**